### PR TITLE
test: Added branch name and PR title checks.

### PR DESCRIPTION
**Pull request by elad-pticha**

### CODE REVIEW

**Comment by elad-pticha** [comment]
**Branch name regex** - should start with any of `feat|fix|refactor|chore|docs|test|library` followed by `/` char.
**Pull request name regex** - should start with any of `feat|fix|refactor|chore|docs|test|library` followed by `:` and a space. Then any char is allowed from in the ABC with also `.`, `_`, `-` and a space.

**Comment by elad-pticha** [comment]
Tracking each commit is not mandatory because we allow only squash commits, so internal commits in the PR will not be a part of the main branch and commit history.

File: .github/workflows/test_pr.yml [hunk 1]
```diff
@@ -24,6 +24,39 @@ jobs:
...
+      - uses: cycodelabs/cimon-action@v0
+        with:
+          prevent: true
+          fail-on-error: true
+      
+      - name: Check branch name
+        env:
+          BRANCH_NAME: ${{ github.head_ref }}
+        run: |
+          branch_prefix="^(feat|fix|refactor|chore|docs|test|library)\/[a-z0-9._-]+$"
```
Comment (oreenlivnicode) [inline .github/workflows/test_pr.yml:40]: What about hotfix?
Also - I prefer "feature" over "feat" but I will leave it for you to decide

Comment (elad-pticha) [reply .github/workflows/test_pr.yml]: Changed according to: https://dev.to/varbsan/a-simplified-convention-for-naming-branches-and-commits-in-git-il4

**Review by alex-ilgayev — APPROVED** [review]

**Pull request merged**